cmd/codegen/generator/typescript/templates/functions.go

```diff
@@ -301,14 +312,51 @@ func (funcs typescriptTemplateFuncs) sortInputFields(s []introspection.InputValu
 
 func (funcs typescriptTemplateFuncs) sortEnumFields(s []introspection.EnumValue) []introspection.EnumValue {
 	slices.SortStableFunc(s, func(x, y introspection.EnumValue) int {
-		return cmp.Compare(funcs.formatEnum(x.Name), funcs.formatEnum(y.Name))
+		return cmp.Compare(strcase.ToCamel(x.Name), strcase.ToCamel(y.Name))
 	})
 	s = slices.CompactFunc(s, func(x, y introspection.EnumValue) bool {
-		return funcs.formatEnum(x.Name) == funcs.formatEnum(y.Name)
+		return strcase.ToCamel(x.Name) == strcase.ToCamel(y.Name)
 	})
+
+	// Remove enums without values that may exist
+	s = slices.DeleteFunc(s, func(x introspection.EnumValue) bool {
+		return x.Name == "" && x.Directives.EnumValue() == ""
```

Comment:
:thinking: how does this happen?

This feels like a bug, I have seen this happen in some edge cases, but some of them should be fixed by https://github.com/dagger/dagger/pull/10647.

The engine should never send values that look like this.

Reply:
Fixed by using `slices.Clone` during array manipulation ;)